src/libos/crates/async-rt/src/time/timer.rs

```diff
@@ -0,0 +1,200 @@
+pub use core::task::Waker as RawWaker;
+
+use crate::prelude::*;
+use crate::time::Instant;
+
+use hierarchical_hash_wheel_timer::wheels::quad_wheel::QuadWheelWithOverflow;
+
+lazy_static! {
+    pub static ref TIMER_WHEEL: TimerWheel = TimerWheel::new();
+}
+
+/// The interface of hierarchical timer wheel.
+/// The resolution of time is 1ms.
```

Comment:
There is a concept called timer slack in Linux.

> The timer slack is the length of time by which the kernel may delay the wake-up of certain system calls that block with a timeout. Permitting this delay allows the kernel to coalesce wake-up events, thus possibly reducing the number of system wake-ups and saving power... The "current" timer slack is used by the kernel to group timer expirations for the calling thread that are close to one another; as a consequence, timer expirations for the thread may be up to the specified number of nanoseconds late. Grouping timer expirations can help reduce system power consumption by minimizing CPU wake-ups... The timer slack values of init (PID 1), the ancestor of all processes, are 50,000 nanoseconds (50 microseconds).

So what is our value of timer slack? Is it 1ms?

Reply:
Well, we don't have the timer slack concept. By since the 1ms resolution, the smallest time unit is 1ms. It mean that if wait for 1 ms and 50 microseconds, we will wait for 2 ms.
The 1ms is the restriction from hierarchical timer wheel crate, since the resolution supported by this crate is 1ms. 
Notes that, 1 ms = 1000 microseconds,  which equal to 20 timer slacks.
If the 1ms is too coarse for ngo, we can impl a timer wheel with higher resolution.